pretty_midi/pretty_midi.py

```diff
@@ -166,27 +168,34 @@ def _load_metadata(self, midi_data):
         """
 
         # Initialize empty lists for storing key signature changes, time
-        # signature changes, and lyrics
+        # signature changes, lyrics and a dictionary (start_time, lyrics index) for faster vocals matching
         self.key_signature_changes = []
         self.time_signature_changes = []
         self.lyrics = []
+        self.__lyrics_times = {}
 
-        for event in midi_data.tracks[0]:
-            if event.type == 'key_signature':
-                key_obj = KeySignature(
-                    key_name_to_key_number(event.key),
-                    self.__tick_to_time[event.time])
-                self.key_signature_changes.append(key_obj)
-
-            elif event.type == 'time_signature':
-                ts_obj = TimeSignature(event.numerator,
-                                       event.denominator,
-                                       self.__tick_to_time[event.time])
-                self.time_signature_changes.append(ts_obj)
-
-            elif event.type == 'lyrics':
-                self.lyrics.append(Lyric(
-                    event.text, self.__tick_to_time[event.time]))
+        # We search for metadata events on all tracks
+        for track in midi_data.tracks:
+            for event in track:
+                if event.type == 'key_signature':
+                    key_obj = KeySignature(
+                        key_name_to_key_number(event.key),
+                        self.__tick_to_time[event.time])
+                    self.key_signature_changes.append(key_obj)
+
+                elif event.type == 'time_signature':
+                    ts_obj = TimeSignature(event.numerator,
+                                           event.denominator,
+                                           self.__tick_to_time[event.time])
+                    self.time_signature_changes.append(ts_obj)
+
+                # Lyrics at time 0 are related to copyright, author, title, etc. so we skip them
+                elif (event.type == 'lyrics' or event.type == 'text') and self.__tick_to_time[event.time] != 0:
+                    # keep (start_time, lyrics index) except for line breaks, that don't correspond to notes
+                    if not event.text.isspace():
+                        self.__lyrics_times[self.__tick_to_time[event.time]] = len(self.lyrics)
+                    self.lyrics.append(Lyric(
+                        event.text, self.__tick_to_time[event.time]))
```

Comment:
I would like you to keep the lyrics processing the same - pretty_midi is not designed to make opinionated decisions about which lyric events should be considered valid (e.g. not at time zero and not newline). These may be useful to some users even if they are not useful to you.

I'd also prefer to store text in a separate self.text_events list, since they are often used for different purposes than lyrics.

Reply:
I was thinking of classifying both lyrics and text events as:
- **text** if they appear at time 0
- **lyrics** if they appear at any other time

since most of the time midi annotators don't distinguish between text and lyrics. What is your opinion on this?